node/bft/src/gateway.rs

```diff
@@ -1119,6 +1126,30 @@ impl<N: Network> Gateway<N> {
     fn handle_banned_ips(&self) {
         self.tcp.banned_peers().remove_old_bans(IP_BAN_TIME_IN_SECS);
     }
+
+    // Update the dynamic validator whitelist.
+    fn update_validator_whitelist(&self) {
+        // Collect the addresses of all the connected validators.
+        let connected_validator_addrs = self.connected_peers();
```

Comment:
We should include validators suggested to us from the bootstrap node, otherwise those can never connect to us

Reply:
You mean the candidates? So we should optimistically whitelist them even if we're not yet sure if they are legitimate?

Reply:
Yep, the verified validators will always come first.